Add suite of flux-corrected transport tests for landice integration

trhille opened the pull request:
Add suite of landice integration tests using flux-corrected transport that is currently being implemented in https://github.com/MALI-Dev/E3SM/pull/70.

<!--
Thank you for your pull request.
Please add a description of what is accomplished in the PR here at the top:
-->

<!--
Below are a few things we ask you or your reviewers to kindly check. 
***Remove checks that are not relevant by deleting the line(s) below.***
-->
Checklist
* [ ] User's Guide has been updated
* [ ] Developer's Guide has been updated
* [ ] API documentation in the Developer's Guide (`api.rst`) has any new or modified class, method and/or functions listed
* [ ] Documentation has been [built locally](https://mpas-dev.github.io/compass/latest/developers_guide/building_docs.html) and changes look as expected
* [ ] Document (in a comment titled `Testing` in this PR) any testing that was used to verify the changes
* [x] New tests have been added to a test suite

<!--
Please note any issues this fixes using closing keywords: https://help.github.com/articles/closing-issues-using-keywords
-->

trhille committed:
Add new fct advection tests for dome group

Add new fct advection tests for dome group

trhille committed:
Fix dome decomposition test namelists for fct

The namelist files in the run step were not being properly updated
to ust fct advection. This fixes that.

trhille committed:
Add greenland fct tests

trhille committed:
Add thwaites fct tests

trhille committed:
Add humboldt fct tests

Add 3km fct full physics tests for humboldt. Note that the decomposition
tests do not pass validation, but testing shows that they also fail
using fo advection.

trhille committed:
Add fct_integration suite

Add fct_integration suite, which uses a subset of the new fct
advection tests based on the tests in full_integration.

matthewhoffman commented on compass/landice/suites/fct_integration.txt:
```diff
@@ -0,0 +1,9 @@
+landice/dome/variable_resolution/fo_fct_restart_test
+landice/greenland/fo_fct_decomposition_test
+landice/greenland/fo_fct_restart_test
+landice/thwaites/fct_decomposition_test
```
I'm wondering if we ought to revamp the naming conventions to make it clear when 'fo' or other abbreviations refer to velocity solve vs advection vs other possible settings.  I imagine `compass list` is getting pretty busy.

trhille replied:
Agreed.

matthewhoffman replied:
Also, making a note that we'll need to change the names of the tests in the full_integration suite (and docs) accordingly (discovered this trying to run full_integration from this branch).

trhille committed:
Fix names of thwaites tests in suite definition file

Fix names of thwaites tests in suite definition file. Note that the
naming convention of these tests is currently {advection}_{velocity_solver},
which is the reverse of the convention for other tests. This will be
addressed in a future commit.

trhille committed:
Use RK2 time integration

Use RK2 time integration with all tests in fct_integration suite.

trhille committed:
Fix indentation error found by CI linter
